applications/GeoMechanicsApplication/custom_elements/U_Pw_base_element.cpp

```diff
@@ -167,40 +162,31 @@ void UPwBaseElement<TDim, TNumNodes>::Initialize(const ProcessInfo& rCurrentProc
     KRATOS_CATCH("")
 }
 
-//----------------------------------------------------------------------------------------
-template <unsigned int TDim, unsigned int TNumNodes>
-void UPwBaseElement<TDim, TNumNodes>::ResetConstitutiveLaw()
+void UPwBaseElement::ResetConstitutiveLaw()
 {
     KRATOS_TRY
 
-    // erasing stress vectors
-    for (unsigned int i = 0; i < mStressVector.size(); ++i) {
-        mStressVector[i].clear();
-    }
+    std::for_each(mStressVector.begin(), mStressVector.end(),
+                  [](auto& stress_vector) { stress_vector.clear(); });
```

Comment:
I agree that for consistency it might be better still to use a range-based `for` loop, rather than `std::for_each`. That STL algorithm was really nice before the range-based `for` loop was available (i.e. prior to C++11). I believe this algorithm has become obsolete now, since it doesn't really add to what you can do with a range-based `for` loop.